OpenSim/Simulation/Model/Model.h

```diff
@@ -457,6 +457,7 @@ OpenSim_DECLARE_CONCRETE_OBJECT(Model, ModelComponent);
     /** Perform computations that may depend on anything but are only used
     for reporting and cannot affect subsequent simulation behavior. **/
     void realizeReport(const SimTK::State& state) const;
+    void extendRealizeVelocity(const SimTK::State& state) const override;
```

Comment:
I know it looks like this is a good place for `extendRealizeVelocity()`, but it really should be somewhere else. The `realizeX()` methods are part of the public interface, and they are to be used by scripting users. For example:

```py
model.realizePosition(state)
com = model.getMassCenter(state)
```

However, `extendRealizeVelocity()` is part of the implementation of the `Model` class, and the method is called by Simbody whenever someone calls `Model::realizeVelocity()`.

So I think this method should be placed somewhere in the protected section, and it should also have a brief comment saying why it's overridden.